### PR TITLE
fix(LabelMaps): Support loading and displaying annotations

### PR DESCRIPTION
**Pull request by scottwittenburg**
Supports loading annotations via url parameter `annotations`.  Example annotations file can be found [here](https://thewtex.github.io/allen-ccf-itk-vtk-zarr/allen_ontology_50.json).

### CODE REVIEW

**Comment by scottwittenburg** [comment]
@thewtex, while this is basically working end-to-end, I think we'll need to iterate on it a little.  For example:

1. I can't remember if the label values correspond to the index in, e.g. `allen_ontology_50.json`, or if it's the `id` values.
2. I show the 'acronym` field, rather than the `name` field, because of how long the `name` seems to be.
3. This does not yet support drag-and-drop annotations files.

File: src/ItkVtkViewProxy.js [hunk 1]
```diff
@@ -93,6 +93,14 @@ function ItkVtkViewProxy(publicAPI, model) {
     }
   }
 
+  function getAnnotationText(value) {
+    const labelValue = value[model.labelIndex]
```
Comment (scottwittenburg) [inline src/ItkVtkViewProxy.js:97]: Should this function live here and the `annotationMap` and `labelIndex` provided to this thing's `model`?  Or should we just provide this proxy class with a method that can be called to retrieve the annotation text?

**Comment by thewtex** [comment]
Looking good, thanks @scottwittenburg! I will take a closer look later on today.

File: src/ItkVtkViewProxy.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ import vtkWidgetManager from 'vtk.js/Sources/Widgets/Core/WidgetManager'
 import * as vtkMath from 'vtk.js/Sources/Common/Core/Math'
 
 const CursorCornerAnnotation =
-  '<table class="corner-annotation" style="margin-left: 0;"><tr><td style="margin-left: auto; margin-right: 0;">Index:</td><td>${iIndex},</td><td>${jIndex},</td><td>${kIndex}</td></tr><tr><td style="margin-left: auto; margin-right: 0;">Position:</td><td>${xPosition},</td><td>${yPosition},</td><td>${zPosition}</td></tr><tr><td style="margin-left: auto; margin-right: 0;"">Value:</td><td>${value}</td></tr></table>'
+  '<table class="corner-annotation" style="margin-left: 0;"><tr><td style="margin-left: auto; margin-right: 0;">Index:</td><td>${iIndex},</td><td>${jIndex},</td><td>${kIndex}</td></tr><tr><td style="margin-left: auto; margin-right: 0;">Position:</td><td>${xPosition},</td><td>${yPosition},</td><td>${zPosition}</td></tr><tr><td style="margin-left: auto; margin-right: 0;"">Value:</td><td>${value}</td></tr><tr><td style="margin-left: auto; margin-right: 0;">Annotation:</td><td>${annotation}</td><td></td><td></td></tr></table>'
```
Comment (thewtex) [inline src/ItkVtkViewProxy.js:13]: Can the *Annotation* row's presence be conditional on the presence of a label map?

File: src/ItkVtkViewProxy.js (hunk 1 above)
Comment (thewtex) [reply src/ItkVtkViewProxy.js:97]: This looks good to me :+1:

File: src/ItkVtkViewProxy.js [hunk 3]
```diff
@@ -93,6 +93,14 @@ function ItkVtkViewProxy(publicAPI, model) {
     }
   }
 
+  function getAnnotationText(value) {
+    const labelValue = value[model.labelIndex]
+    if (model.annotationMap != null && model.annotationMap.has(labelValue)) {
+      return model.annotationMap.get(labelValue)
+    }
+    return 'Unknown'
```
Comment (thewtex) [inline src/ItkVtkViewProxy.js:101]: `labelValue` string is preferred.

File: src/createViewer.js [hunk 4]
```diff
@@ -23,6 +23,7 @@ const createViewer = (
     multiscaleImage,
     labelMap,
     multiscaleLabelMap,
+    annotations,
```
Comment (thewtex) [inline src/createViewer.js:26]: -> `labelMapAnnotations` since geometries could have lookup tables.

File: src/processFiles.js [hunk 5]
```diff
@@ -176,6 +187,14 @@ export const readFiles = async ({
       webWorker.terminate()
       labelMapData = vtkITKHelper.convertItkToVtkImage(itkImage)
     }
+    let annotationData = null
+    if (!!annotations) {
+      annotationData = new Map(
+        annotations
+          .filter(item => !!item)
+          .map(({ acronym }, idx) => [idx, acronym])
```
Comment (thewtex) [inline src/processFiles.js:195]: -> factor out for generality.

**Comment by scottwittenburg** [comment]
@thewtex Mind taking another look when you get a chance?  I think I addressed all the concerns you raised, and if we can get it merged, I would like to make use of some of this work in #282.

Thanks!

**Pull request merged**